src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -132,26 +136,102 @@
             (str "Unable to find bootstrap.cfg file via --bootstrap-config "
                  "command line argument, current working directory, or on classpath")))))
 
-(schema/defn chain-files :- [schema/Str]
-  "Takes a list of files, reads all their lines in, and returns a flattened seq
-   of all their lines put together.
+(schema/defn indexed
+  "Returns seq of [index, item] pairs
+  [:a :b :c] -> ([0 :a] [1 :b] [2 :c])"
+  [coll]
+  (map vector (iterate inc 0) coll))
 
-   In the case of an empty file, nil would be returned by line-seq, so we remove
-   all the nils at the end"
-  [files :- [(schema/protocol io/IOFactory)]]
-  (remove nil? (flatten (map #(line-seq (io/reader %)) files))))
+(schema/defn get-annotated-bootstrap-entries :- [AnnotatedBootstrapEntry]
+  [configs :- [(schema/protocol io/IOFactory)]]
+  (flatten (for [config configs]
+             (for [[line-number line-text] (indexed (map remove-comments (line-seq (io/reader config))))
```

Comment:
It might be worth adding a couple of unit tests that just cover this function directly, including one that has a file with a commented line in it; that'd probably make it a little more obvious.